generic_domains.php

```diff
@@ -144,6 +145,69 @@ public function editService($package, $service, array $vars = [], $parent_packag
         return $this->addService($package, $vars, $parent_package, $parent_service);
     }
 
+    /**
+     * Returns all fields to display to an admin attempting to add a service with the module.
+     *
+     * @param stdClass $package A stdClass object representing the selected package
+     * @param $vars stdClass A stdClass object representing a set of post fields
+     * @return ModuleFields A ModuleFields object, containing the fields to render
+     *  as well as any additional HTML markup to include
+     */
+    public function getAdminAddFields($package, $vars = null)
+    {
+        $fields = Configure::get('GenericDomains.domain_fields');
```

Comment:
I think "GenericDomains.transfer_fields" should be used here, as the EPP code can get lost in some scenarios. For example, if someone initiates a domain transfer through the Order plugin, the service is marked as pending, and if I decide to activate it manually, because the EPP field is not present, it doesn't saved and it gets lost.

Reply:
Updated